### ISSUE TITLE
iaas-configuration in `staged-director-config` does not contain all properties

### ISSUE THREAD

**Issue by kcboyle** [pivotal-cf/om#220]
more detail can be found here: https://github.com/pivotal-cf/om/issues/192

There are params that the OpsManager API does not return when getting back the `iaas-configuration`

Take vsphere as an example, these are all the configurable proerties (required to configure-director):
```
"vcenter_host":                 "((iaas-configuration_vcenter_host))",
"datacenter":                   "((iaas-configuration_datacenter))",
"ephemeral_datastores_string":  "((iaas-configuration_ephemeral_datastores_string))",
"persistent_datastores_string": "((iaas-configuration_persistent_datastores_string))",
"vcenter_username":             "((iaas-configuration_vcenter_username))",
*****"vcenter_password":             "((iaas-configuration_vcenter_password))",*****
"bosh_vm_folder":               "((iaas-configuration_bosh_vm_folder))",
"bosh_template_folder":         "((iaas-configuration_bosh_template_folder))",
"bosh_disk_path":               "((iaas-configuration_bosh_disk_path))",
"ssl_verification_enabled":     "((iaas-configuration_ssl_verification_enabled))",
"nsx_networking_enabled":       "((iaas-configuration_nsx_networking_enabled))",
These are the properties we get back from a vsphere opsman API:
```
```
{
"vcenter_host":                 "((iaas-configuration_vcenter_host))",
"datacenter":                   "((iaas-configuration_datacenter))",
"ephemeral_datastores_string":  "((iaas-configuration_ephemeral_datastores_string))",
"persistent_datastores_string": "((iaas-configuration_persistent_datastores_string))",
"vcenter_username":             "((iaas-configuration_vcenter_username))",
"bosh_vm_folder":               "((iaas-configuration_bosh_vm_folder))",
"bosh_template_folder":         "((iaas-configuration_bosh_template_folder))",
"bosh_disk_path":               "((iaas-configuration_bosh_disk_path))",
"ssl_verification_enabled":     "((iaas-configuration_ssl_verification_enabled))",
"nsx_networking_enabled":       "((iaas-configuration_nsx_networking_enabled))",
}
```

If we wanted to include `vcenter_password`, for example, we'd have to have some sort of map that includes certain properties for certain IaaSs on potetentially different versions of OpsMan (if params changed). This is being done in platform-automation right now, but should the logic for this live inside of `om`?

From @mcwumbly: 
```
Ideally, we'd get this back from Ops Manager. This feels like something 
we request from them, andpotentially have backported. In the meantime, 
I'm not sure what the bet thing to do is. I can see it both ways.

If you're going to maintain this map anyway, I'm not sure it matters whether
 that code lives in om or some wrapper. There may be a case to bring it into om. 
```

Opening this issue to discuss

**Comment by mcwumbly** [comment]
Thanks for creating this issue @kcboyle. 

Even if we have a strong case for getting this into the Ops Manager API eventually, let's explore first what this would look like if this responsibility were to live in `om` alone.

1. What would this look like in `om`?
    Can you add another comment here illustrating what these maps would look like based on the work you've already done in your platform-automation tools?  I'd be curious to see a complete example for each IAAS you're currently supporting.

2. What would it look like to support multiple versions of Ops Manager?
     Have you already done that for older versions?  If not, maybe a brief exploration would help us assess whether there are any meaningful differences in the past ~3 versions of OM? We may be able to guess from that what the maintenance burden would look like going forwards.

**Comment by kcboyle** [comment]
In `om`, this would likely be something that we had to maintain per-iaas and (potentially) per-version of OpsManager. After multiple conversations with the Ops Manager team, we're not getting a lot of hope for these missing fields being included in the OpsMan API any time soon. 

1. One way to do this would be to run `staged-director-config`, key off of some unique field in the iaas (that allowed us to know what iaas the command was running from) and inject the missing fields into the resulting yaml that the command would output (we can find those missing fields via the API docs for OpsMan, and that was their desired solution to the problem).
The maps in our platform automation tool were buggy, and did not honestly answer the issue. (for an example of what we did, see this [commit](https://github.com/pivotal/platform-automation/blob/63aaef996d95e851316e139d2472c17b619ae231/pkg/commands/staged_director_config.go) A reason why it is buggy is that we just hard coded the entire `iaas-configuration`, so any options a user had selected in the UI were ignored (not just the secret ones). So trying to find a hybrid would be optimal for this use case. It would likely be a lot of work on `om`'s part, and might not be the cleanest solution, but if we cannot access those hidden fields any way but by perusing the [OpsManager API docs](https://docs.pivotal.io/pivotalcf/2-2/opsman-api/#updating-director-and-iaas-properties-experimental) -- see the example requests on the right, there might not be a better solution. 

2. We have not figured out how to support multiple versions of Ops Manager, should there be any changes to these sensitive fields. Ideally, the sensitive fields would be the only ones that `om` would have to maintain on differing versions of OpsManager (if a hybrid solution mentioned above could be accomplished). From 2.0->now, there have been no changes to these fields, so should be a minimal maintenance burden on `om` itself.
